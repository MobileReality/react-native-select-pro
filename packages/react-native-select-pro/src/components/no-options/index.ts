export { NoOptions } from './no-options';
