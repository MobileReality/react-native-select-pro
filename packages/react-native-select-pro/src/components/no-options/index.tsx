import React, { ComponentProps } from 'react';
import { StyleSheet, Text, TextStyle, View, ViewStyle } from 'react-native';

import { COLORS, FONT_SIZE, PADDING } from '../../constants/styles';
import type { OptionalToRequired } from '../../helpers/types/OptionalToRequired';
import type { OptionsList } from '../options-list';

type FromParentProps = Pick<
    ComponentProps<typeof OptionsList>,
    'noOptionsText'
>;

type NoOptionsProps = OptionalToRequired<FromParentProps>;

export const NoOptions = ({ noOptionsText }: NoOptionsProps) => {
    return (
        <View style={styles.option}>
            <Text style={styles.text}>{noOptionsText}</Text>
        </View>
    );
};

type Styles = {
    option: ViewStyle;
    text: TextStyle;
};

export const styles = StyleSheet.create<Styles>({
    option: {
        padding: PADDING,
    },
    text: {
        fontSize: FONT_SIZE,
        color: COLORS.BLACK,
        textAlign: 'left',
    },
});
