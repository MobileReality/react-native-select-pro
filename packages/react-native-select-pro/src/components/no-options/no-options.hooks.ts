import { useOptionsListContext } from '../../context';

export const useNoOptions = () => {
    const {
        noOptionsText,
        noOptionsProps,
        noOptionsTextProps,
        styles: mainStyles,
    } = useOptionsListContext();

    const { noOptions } = mainStyles ?? {};
    const { text: textCustomStyles, container: containerCustomStyles } = noOptions ?? {};

    return {
        noOptionsText,
        noOptionsProps,
        noOptionsTextProps,
        textCustomStyles,
        containerCustomStyles,
    };
};
