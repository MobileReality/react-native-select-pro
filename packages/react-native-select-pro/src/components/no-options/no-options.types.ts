import type { ComponentProps } from 'react';

import type { OptionalToRequired } from '../../helpers';
import type { OptionsList } from '../options-list';

type FromParentProps = Pick<ComponentProps<typeof OptionsList>, 'noOptionsText'>;

export type NoOptionsProps = OptionalToRequired<FromParentProps>;
