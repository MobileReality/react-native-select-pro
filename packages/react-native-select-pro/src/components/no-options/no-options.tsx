import React from 'react';
import type { TextStyle, ViewStyle } from 'react-native';
import { StyleSheet, Text, View } from 'react-native';

import { COLORS, FONT_SIZE, PADDING } from '../../constants';

import { useNoOptions } from './no-options.hooks';

export const NoOptions = () => {
    const {
        noOptionsText,
        noOptionsProps,
        noOptionsTextProps,
        textCustomStyles,
        containerCustomStyles,
    } = useNoOptions();

    return (
        <View {...noOptionsProps} style={[styles.option, containerCustomStyles]}>
            <Text {...noOptionsTextProps} style={[styles.text, textCustomStyles]}>
                {noOptionsText}
            </Text>
        </View>
    );
};

type Styles = {
    option: ViewStyle;
    text: TextStyle;
};

export const styles = StyleSheet.create<Styles>({
    option: {
        padding: PADDING,
    },
    text: {
        fontSize: FONT_SIZE,
        color: COLORS.BLACK,
        textAlign: 'left',
    },
});
