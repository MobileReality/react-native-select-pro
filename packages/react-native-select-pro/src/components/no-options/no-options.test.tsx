import React from 'react';

import { NoOptions } from './no-options';

describe('NoOptions', () => {
    it('should generate NoOptions snapshot', () => {
        const noOptions = <NoOptions noOptionsText="Test" />;
        expect(noOptions).toMatchSnapshot();
    });
});
