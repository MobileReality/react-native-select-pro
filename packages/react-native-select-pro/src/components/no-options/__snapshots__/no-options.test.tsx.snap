// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoOptions should generate NoOptions snapshot 1`] = `
<NoOptions
  noOptionsText="Test"
/>
`;
